server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/idp/IdpInitiatedLoginController.java

```diff
@@ -119,6 +119,10 @@ public void initiate(@RequestParam(value = "sp", required = false) String sp,
 
     public String getAssertionConsumerURL(String sp) throws MetadataProviderException {
         EntityDescriptor entityDescriptor = metadataManager.getEntityDescriptor(sp);
+        if (entityDescriptor == null) {
+            throw new MetadataProviderException("Entity description not found for service provider " + sp);
```

Comment:
Typo.  Should be "Entity descriptor not found..."

Change message to "Entity descriptor not found in metadata for service provider... ".  That points consumer to where to fix the problem.

Also, service provider name is usually a long string, something like "uaa-acceptance-zone.cloudfoundry-saml-login".  To make the exception message more readable, include SP name into []

"Entity descriptor not found in metadata for service provider []"